src/transactions/TransactionFrame.cpp

```diff
@@ -795,6 +797,31 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
                 return false;
             }
 
+            if (gIsProductionNetwork && ledgerSeq == 34793621)
```

Comment:
I like this approach, but I don't think this is in the right place. Specifically, this won't generate the correct meta when replaying this ledger. The meta should show that this sponsorship was added to the offer, in the specific transaction where that actually happened and no others. Probably the best thing to do here is push this into `convertWithOffers`, since at that point you can see if this particular offer was actually crossed.

Reply:
fixed.